### PR TITLE
Utilize a submake structure to unify the build tree

### PR DESCRIPTION
**Pull request by RyanHir**
The goal of this PR is to merge all the build scripts for each OS into one as the majority of the scripts are equal. The changes I made utilizes make and extends upon a core Makefile for each OS to apply any changes that are OS specific. Within the shell scripts, there are enviorment variables to apply different flags, for example, the script will check if the target OS is windows and will apply any expat flags to the gdb build.

By having the build system use one set of scripts, maintainability becomes easier as changes are made in the future.

Some things that were not a priority is to change the build flow and how the scripts prioritize tasks with the make build server, or to convert the build scripts to pure make targets.

While the current state of this PR works, it is not pretty and any requests or changes are welcome.

In the case of a merge, it would possibly break other pending PR requests such as #22 until they are fixed to support the new build tree.

### CODE REVIEW

File: makes/src/expat [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/zsh
```
Comment (auscompgeek) [inline makes/src/expat]: Is this intended to be zsh?

Comment (RyanHir) [reply makes/src/expat]: I believe that was simply a artifact from the windows makes. shellcheck appears to be happy with /bin/sh so I will replace it and wait for the CI to complete.

Comment (auscompgeek) [reply makes/src/expat]: I think you missed a couple.

Comment (RyanHir) [reply makes/src/expat]: Seems like expat silently used a zsh feature that looked like a bash feature, should *hopefully* be fixed now.

As for the other files I left everything the same except for in the makes folder

**Pull request closed without merging**